Update CI.yml

Drvanon opened the pull request:
The old branch was no longer in use so for this to be here does not make much sense.

Drvanon committed:
Update CI.yml

marickmanrho commented on .github/workflows/CI.yml:
```diff
@@ -7,9 +7,9 @@ name: CI
 on:
   push:
-    branches: [ CI ]
+    branches: [ master, api-development ]
```
This should be changed to development,

Drvanon replied:
Good call.

Drvanon committed:
Update CI.yml

We would like to make absolutely sure that master is testing positive and pushes are rare, so this is not costly. Pull requests to development are what is most important, but when they succeed that will be the new state of the development branch, so we won't need to test that as well.

marickmanrho reviewed (approved):
Great, ready to merge.
